parachain/pallets/basic-channel/src/outbound/mod.rs

```diff
@@ -215,56 +246,101 @@ pub mod pallet {
 			Ok(())
 		}
 
+		/// Commit messages enqueued on the outbound channel.
+		/// For every account id in the enqueued messages, create a message bundle containing the
+		/// messages for that account. Hash the ethabi encoding of these message bundles in a
+		/// Merkle tree to produce a commitment hash. Then:
+		/// - Store the commitment hash on the parachain for the Ethereum light client to query.
+		/// - Emit an event with the commitment hash and SCALE-encoded message bundles for a
+		/// relayer to read.
+		/// - Persist the ethabi-encoded message bundles to off-chain storage.
 		fn commit() -> Weight {
-			let messages = <MessageQueue<T>>::take();
-			if messages.is_empty() {
+			// TODO: consider using mutate_exists here. If some part of emitting message bundles
+			// fails, we don't want the MessageQueue to be empty.
+			let message_queue = <MessageQueue<T>>::take();
```

Comment:
I'm still concerned about this, but the only failure case I can see is nonce overflow when creating the message bundles. Is this actually something we should be concerned about?

Reply:
Created an [issue](https://linear.app/snowfork/issue/SNO-310) to track this